python/test/result_test.py

```diff
@@ -244,5 +249,115 @@ def test_load_result_with_none(self):
         result.set_score_aggregate_method(ListStats.nonemean)
         self.assertAlmostEqual(result['STRRED_feature_srred_score'], 5829.2644469999996, places=4)
 
+class ResultAggregatingTest(unittest.TestCase):
+
+    def test_invalid_format_aggregator(self):
+
+        print 'test on running result aggregation with invalid format...'
+        with self.assertRaises(AssertionError):
+            XML_Assembler(['dummy.png']).assemble()
+            XML_Assembler(['dummy.png, dummpy.xml']).assemble()
+            JSON_Assembler(['dummy.png']).assemble()
+            JSON_Assembler(['dummy.xml, dummpy.png']).assemble()
+
+    def test_xml_aggregator(self):
+
+        print 'test on running XML aggregation...'
+        ref_path, dis_path, asset, asset_original = set_default_576_324_videos_for_testing()
+
+        asset_list = [asset, asset_original]
+
+        self.runner = VmafQualityRunner(
+            asset_list,
+            None, fifo_mode=True,
+            delete_workdir=True,
+            result_store=None,
+            optional_dict={
+                'model_filepath': VmafConfig.model_path("vmaf_v0.6.1.pkl"),
+            },
+            optional_dict2=None,
+        )
+        self.runner.run()
+
+        results = self.runner.results
+
+        xml_file_list = []
+        dump_dir = os.getcwd() + '/temp_outputs/'
+        make_parent_dirs_if_nonexist(dump_dir)
+
+        xml_file = dump_dir + 'temp_xml_1.xml'
+        with open(xml_file, 'wb') as myfile:
+            myfile.write(results[0].to_xml())
+        xml_file_list.append(xml_file)
+
+        xml_file = dump_dir + 'temp_xml_2.xml'
+        with open(xml_file, 'wb') as myfile:
+            myfile.write(results[1].to_xml())
+        xml_file_list.append(xml_file)
+
+        import xml.etree.ElementTree as ET
+        tree_1 = ET.ElementTree(ET.fromstring(results[0].to_xml()))
+        tree_2 = ET.ElementTree(ET.fromstring(XML_Assembler([xml_file_list[0]]).assemble()))
+
+        tree_3 = ET.ElementTree(ET.fromstring(XML_Assembler(xml_file_list).assemble()))
+        tree_4 = ET.ElementTree(ET.fromstring(XML_Assembler([xml_file_list[1], xml_file_list[0]]).assemble()))
```

Comment:
This is to test if the order of the files in the list changes, the result doesn't change?

Once assembled, will you re-serialize the frame numbers? You should. And make sure to add a test to test it.